plugins/package-managers/node/src/main/kotlin/yarn/Yarn.kt

```diff
@@ -107,11 +107,16 @@ class Yarn(override val descriptor: PluginDescriptor = YarnFactory.descriptor) :
         val workspaceModuleDirs = getWorkspaceModuleDirs(workingDir)
         handler.setContext(workingDir, getModuleDirs(workingDir), workspaceModuleDirs)
 
-        val scopes = Scope.entries
+        val scopes = Scope.entries.filterNot { scope -> scope.isExcluded(excludes) }
         val moduleInfosForScope = scopes.associateWith { scope ->
             listModules(workingDir, scope).resolveVersions().undoDeduplication()
         }
 
+        // Warm-up the cache to speed-up processing.
+        getAllModuleIds(moduleInfosForScope.values.flatten()).let { moduleIds ->
+            handler.requestAllPackageDetails(moduleIds)
+        }
+
```

Comment:
This part (and the function added below) look like they should have gone to a separate commit.